src/AliasStorage.php

```diff
@@ -46,8 +46,12 @@ public function setDomainId($domain_id) {
   public function getDomainId() {
     // If no domain id has been set, use the currently active one.
     if (is_null($this->domain_id)) {
-      $this->domain_id = (int) $this->domain_negotiator->getActiveDomain()->getDomainId();
+      if (!is_null($this->domain_negotiator->getActiveDomain())) {
+        $this->domain_id = (int) $this->domain_negotiator->getActiveDomain()
```

Comment:
The alias should always be looked up for the current domain unless being overridden with setDomainId() when saving. Perhaps we can set the current domain using setDomainId() where required, but this might be in many places.